src/crisptrutski/boot_cljs_test.clj

```diff
@@ -27,145 +23,171 @@
 
 (defn- gen-suite-ns
   "Generate source-code for default test suite."
-  [ns test-namespaces]
-  (let [ns-spec `(~'ns ~ns (:require [doo.runner :refer-macros [~'doo-tests ~'doo-all-tests]]
-                                     ~@(mapv vector test-namespaces)))
-        run-exp `(~'doo-tests ~@(map u/normalize-sym test-namespaces))]
-    (->> [ns-spec run-exp]
+  [ns namespaces]
+  (let [ns-spec `(~'ns ~ns (:require [doo.runner :refer-macros [~'doo-tests]] ~@(mapv vector namespaces)))
+        run-exp `(~'doo-tests ~@(map u/normalize-sym namespaces))]
+    (->> [ns-spec '(enable-console-print!) run-exp]
          (map #(with-out-str (clojure.pprint/pprint %)))
          (str/join "\n" ))))
 
 (defn add-suite-ns!
   "Add test suite bootstrap script to fileset."
-  [fileset tmp-main suite-ns test-namespaces]
+  [fileset tmp-main id namespaces]
   (ensure-deps! [:adzerk/boot-cljs])
-  (let [out-main (u/ns->cljs-path suite-ns)
-        out-file (doto (io/file tmp-main out-main) io/make-parents)
-        out-path (u/relativize (.getPath tmp-main) (.getPath out-file))
-        cljs     (u/cljs-files fileset)]
-    (if (contains? (into #{} (map core/tmp-path) cljs) out-path)
-      (do (info "Using %s...\n" out-path)
-          fileset)
-      (do (info "Writing %s...\n" out-path)
-          (spit out-file (gen-suite-ns suite-ns test-namespaces))
-          (-> fileset (core/add-source tmp-main) core/commit!)))))
+  (let [relative   #(u/relativize (.getPath tmp-main) (.getPath %))
+        out-main   (str id ".cljs")
+        src-file   (doto (io/file tmp-main out-main) io/make-parents)
+        edn-file   (io/file tmp-main (str out-main ".edn"))
+        src-path   (relative src-file)
+        edn-path   (relative edn-file)
+        exists?    (into #{} (map core/tmp-path) (u/cljs-files fileset))
+        suite?     (or (exists? src-path) (exists? (str/replace src-path ".cljs" ".cljc")))
+        edn?       (exists? edn-path)
+        edn        (when edn? (read-string (slurp (core/tmp-file (core/tmp-get fileset edn-path)))))
+        ;; subset namespaces to those directly required by .cljs.edn
+        ;; TODO: better would be to use transitive requires, once we have the dep graph
+        namespaces (if edn (filter (set (:require edn)) namespaces) namespaces)
```

Comment:
@Deraen one part I'm not super happy about, but better than errors trying to run a namespace that hasn't loaded